client/Hooks/useRouteConfig/useRouteConfig.hook.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+import { RouterConfig, PageConfig, Page, Leaf } from 'Hooks/types';
+
+const isTopLevelNav = (path) => path.match(/\//g).length === 1;
+
+const extractTopLevelPath = (path) => path.match(/\/[^/]*/g)[0];
+
+const generateKey = (name, prefix, properties) =>
+  `${prefix}-${removeWhitespace(removePathVars(name))}${
+    properties && properties.mode ? `.${properties.mode}` : ''
+  }`;
+
+const removeWhitespace = (name) => name.replace(/ /g, '');
+
+const removePathVars = (name) => name.replace(/\${([^}]*)\}/g, '');
+
+const useRouteConfig = (input: PageConfig): RouterConfig => {
+  const output: RouterConfig = {
+    links: [],
+    routes: [],
+    meta: {},
+  };
+
+  const leaves: Array<Leaf> = [];
+
+  Object.values(input).map((value) => {
+    const page = value as Page;
+    const contexts = page.contexts;
+
+    contexts.map((context) => {
```

Comment:
being uber picky, wouldn't a context reduce (ie take array of contexts, reduce them to a single `output` shape) be cleaner? Rather than mapping and changing output by scope?

Reply:
Will look into doing that 👍

Reply:
Or alternatively:
```
Object.values(input).reduce((total, page) => [...total, ...page.contexts])
```